ui/src/pages/login/Login.tsx

```diff
@@ -234,10 +201,12 @@ const PartiesLoginForm: React.FC<Props> = ({ onLogin }) => {
     }
   };
 
-  const handleLoad = async (parties: PartyDetails[]) => {
-    setParties(parties);
-    setSelectedPartyId(parties[0]?.party || '');
-    storeParties(parties);
+  const handleLoad = async (parties: PartyToken[]) => {
```

Comment:
If you want, you could re-use some of this on line 188. Looks like you check and set parties + selectedPartyId in the same way